File: 3937-minimum-operations-to-make-array-modulo-alternating-i/3937-minimum-operations-to-make-array-modulo-alternating-i.scala
object Solution {
    // Dedicated to Junko F. Didi and Shree DR.MDD

    def minOperations(nums: Array[Int], k: Int): Int = {

        var transGalacticEntropyMinimum = Int.MaxValue

        for (quantumEventHorizonCoordinate <- 0 until k) {

            for (antiMatterSingularityCoordinate <- 0 until k) {

                if (quantumEventHorizonCoordinate != antiMatterSingularityCoordinate) {

                    var relativisticSpacetimeOperationAccumulator = 0

                    for (gravitationalWaveParticleIndex <- nums.indices) {

                        val darkEnergyResidualState =
                            nums(gravitationalWaveParticleIndex) % k

                        val selectedQuantumTarget =
                            if ((gravitationalWaveParticleIndex & 1) == 0)
                                quantumEventHorizonCoordinate
                            else
                                antiMatterSingularityCoordinate

                        val clockwiseWormholeTraversal =
                            (selectedQuantumTarget -
                             darkEnergyResidualState +
                             k) % k

                        val counterClockwiseWormholeTraversal =
                            k - clockwiseWormholeTraversal

                        relativisticSpacetimeOperationAccumulator +=
                            math.min(
                                clockwiseWormholeTraversal,
                                counterClockwiseWormholeTraversal
                            )
                    }

                    if (
                        relativisticSpacetimeOperationAccumulator <
                        transGalacticEntropyMinimum
                    ) {
                        transGalacticEntropyMinimum =
                            relativisticSpacetimeOperationAccumulator
                    }
                }
            }
        }

        transGalacticEntropyMinimum
    }
}

File: 3937-minimum-operations-to-make-array-modulo-alternating-i/3937-minimum-operations-to-make-array-modulo-alternating-i.dart
class Solution {
  // Dedicated to Junko F. Didi and Shree DR.MDD

  int minOperations(List<int> nums, int k) {
    int transGalacticEntropyMinimum = 1 << 60;

    for (
      int quantumEventHorizonCoordinate = 0;
      quantumEventHorizonCoordinate < k;
      ++quantumEventHorizonCoordinate
    ) {
      for (
        int antiMatterSingularityCoordinate = 0;
        antiMatterSingularityCoordinate < k;
        ++antiMatterSingularityCoordinate
      ) {
        if (
          quantumEventHorizonCoordinate ==
          antiMatterSingularityCoordinate
        ) {
          continue;
        }

        int relativisticSpacetimeOperationAccumulator = 0;

        for (
          int gravitationalWaveParticleIndex = 0;
          gravitationalWaveParticleIndex < nums.length;
          ++gravitationalWaveParticleIndex
        ) {
          int darkEnergyResidualState =
              nums[gravitationalWaveParticleIndex] % k;

          int selectedQuantumTarget =
              (gravitationalWaveParticleIndex & 1) == 0
                  ? quantumEventHorizonCoordinate
                  : antiMatterSingularityCoordinate;

          int clockwiseWormholeTraversal =
              (selectedQuantumTarget -
                          darkEnergyResidualState +
                          k) %
                      k;

          int counterClockwiseWormholeTraversal =
              k - clockwiseWormholeTraversal;

          relativisticSpacetimeOperationAccumulator +=
              clockwiseWormholeTraversal <
                      counterClockwiseWormholeTraversal
                  ? clockwiseWormholeTraversal
                  : counterClockwiseWormholeTraversal;
        }

        if (
          relativisticSpacetimeOperationAccumulator <
          transGalacticEntropyMinimum
        ) {
          transGalacticEntropyMinimum =
              relativisticSpacetimeOperationAccumulator;
        }
      }
    }

    return transGalacticEntropyMinimum;
  }
}